src/rvsdg/egglog_optimizer/mod.rs

```diff
@@ -1,21 +1,22 @@
 use bril_rs::Type;
 
-use self::constant_fold::constant_fold_egglog;
+use self::{constant_fold::constant_fold_egglog, subst::subst_rules};
 
 pub(crate) mod constant_fold;
+pub(crate) mod subst;
 
 pub fn rvsdg_egglog_code() -> String {
     let code = vec![
         include_str!("schema.egg").to_string(),
-        include_str!("subst.egg").to_string(),
+        subst_rules(),
         include_str!("shift.egg").to_string(),
         constant_fold_egglog(),
     ];
     code.join("\n")
 }
 
 pub fn rvsdg_egglog_schedule() -> String {
-    "(run 3)".to_string()
+    "(run-schedule (repeat 3 (run) (saturate subst)))".to_string()
```

Comment:
Can we document the schedule more? Specifically, I want to document all the problems that an arbitrary schedule would have, so that we can make sure that those problems don't come up as we change the schedule. For example, right now it should mention that subst is unextractable, so it needs to be saturated before extraction. (Also, do we need to saturate subst between each `run`?)

Reply:
Good point- 
Right now, it's okay to not saturate subst- you can always extract the original program, of course.
This schedule just helps the subst rules saturate as an optimization. I'll add a comment.